### ISSUE TITLE
Investigate how to make the build flow works after generating an ill-formed IDL

### ISSUE THREAD

**Issue by roxk** [roxk/idlgen#60]
If for whatever reasons idlgen generated a mal-formed IDL, the bootstrap phase would fail. Then, even though devs had later fixed the mistake in header, idlgen cannot generate new, correct IDL as it's blocked by bootstrap.

The IDL can be ill-formed due to lots of reasons, e.g. missing import, etc. There is no way idlgen could detect all of these issues without doing a complete MIDL language analysis, which is what the tool wants to avoid.

Ideas:
- Remove generated .idl before bootstrap?

**Comment by roxk** [comment]
With the ability to skip the current header's own projection, asking user to clear the idl is an acceptable workaround. Will close.

**Comment by roxk** [comment]
Actually, we might be able to help devs empty the idl before bootstrapping if an IDL needs to be generated. But then `IdlgenCppMidl`'s input would have to include C++ headers as well for stale-detection.

**Comment by roxk** [comment]
Since the input should only be what is included in `IdlgenCppIncludes`, using another task is easier. `IdlgenCppMidl` could then depend on this task.

**Comment by roxk** [comment]
Now that this is merged, README can omit the entire section on troubleshooting except for bugs/as a workaround.
